presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -177,10 +182,10 @@ public NodeSelector createNodeSelector(Session session, ConnectorId connectorId,
 
             Set<InternalNode> nodes;
             if (connectorId != null) {
-                nodes = nodeManager.getActiveConnectorNodes(connectorId);
+                nodes = resilientSoftAffinityEnabled ? new LinkedHashSet<>(nodeManager.getAllConnectorNodes(connectorId)) : new LinkedHashSet<>(nodeManager.getActiveConnectorNodes(connectorId));
```

Comment:
Are we able to get node selection strategy here to help with node selection? If we have `resilientSoftAffinityEnabled` as `true` but a specific connector does not support soft affinity, will it be an issue (such as, it gets all nodes, and randomly picked a dead node because it is the least loaded one)?

One other option is to always return both active and all, and let node selector choose nodes based on node selection strategy.

Reply:
Makes sense. I tried separating out allnodes and activenodes in my latest push.